packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -315,23 +323,26 @@ class _WindowsUtils extends OperatingSystemUtils {
 
   @override
   void zip(Directory data, File zipFile) {
-    final Archive archive = Archive();
-    for (final FileSystemEntity entity in data.listSync(recursive: true)) {
-      if (entity is! File) {
-        continue;
-      }
-      final File file = entity as File;
-      final String path = file.fileSystem.path.relative(file.path, from: data.path);
-      final List<int> bytes = file.readAsBytesSync();
-      archive.addFile(ArchiveFile(path, bytes.length, bytes));
-    }
-    zipFile.writeAsBytesSync(ZipEncoder().encode(archive), flush: true);
+    _processManager.runSync(<String>[
+      _activePowershell,
+      'Compress-Archive',
+      '-Path',
+      data.path,
+      '-DestinationPath',
+      zipFile.path,
+    ]);
   }
 
   @override
   void unzip(File file, Directory targetDirectory) {
-    final Archive archive = ZipDecoder().decodeBytes(file.readAsBytesSync());
-    _unpackArchive(archive, targetDirectory);
+    _processManager.runSync(<String>[
+      _activePowershell,
+      'Expand-Archive',
+      '-Path',
+      file.path,
+      '-DestinationPath',
+      targetDirectory.path,
+    ]);
   }
 
   @override
```

Comment:
If `Expand-Archive` fails when the zip file is malformed, I wonder if `verifyZip` could be implemented by using the Windows equivalent of `/dev/null` as the output path.

Reply:
TIL about `NUL` https://stackoverflow.com/questions/313111/is-there-a-dev-null-on-windows

Reply:
Looking at cache.dart, I'm not really sure why we're verifying the zip. Expand-Archive doesn't extract anything if it fails, so verification seems like we would just be extracting it twice